textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/regexmatch/RegexToGramQueryTranslatorTest.java

```diff
@@ -17,133 +17,321 @@ private void printTranslatorResult(String regex) {
 		
 		System.out.println("regex: "+regex);
 		System.out.println("boolean expression: "+exactQuery.getLuceneQueryString());
+		
 		System.out.println("query tree: ");
 		System.out.println(exactQuery.printQueryTree());
+		
+		GramBooleanQuery dnf = GramBooleanQuery.toDNF(exactQuery);
+		GramBooleanQuery simplifiedDNF = GramBooleanQuery.simplifyDNF(dnf);
+		
+		System.out.println("Simplified DNF: ");
+		System.out.println(simplifiedDNF.printQueryTree());
+
 		System.out.println();
 	}
 	
 	@Test
 	public void testEmptyRegex() {
-		GramBooleanQuery exactQuery = RegexToGramQueryTranslator.translate("");
+		String regex = "";
+		
+		GramBooleanQuery exactQuery = RegexToGramQueryTranslator.translate(regex);
```

Comment:
All the test cases have been automated. Before transforming to DNF, the query is too complicated for us to write tests, but after transforming and simplifications, we are able to write the expected result for all test cases.